module-2/backend/week_two.md

```diff
@@ -26,20 +32,65 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
+Team.find(4).owner
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+There will be many students to one teacher.  The students will have a foreign key associated with their teacher.
+
 6. Define foreign key, primary key, and schema.
+
+A foreign key acts as a placeholder for data from a table that is related to the table that contains the foreign key (IE, the teacher ID instead of any other information about the teacher).  The primary key is the key used to identify a particular row of data (the teacher's teacher ID that exists on the teachers table).  The schema is a way of describing how data is organized in each table.  The schema contains information like what the column names are and what type of data each colum contains.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+A foreign key represents a primary key of another table and is used to form a relationship between two tables.  The foreign key will refer to the primary key of the table that is being referenced.  For example, each row in the teacher's table has an ID, but each row in the students tables has a teacher_id which is used to relate the two datasets.
+
 8. What are the parts of an HTTP response?
 
+A request and response.
```

Comment:
You could argue that these are the high level parts of the HTTP request/response cycle, but a response has a

* status line
* headers
* body